src/spi.rs

```diff
@@ -482,70 +548,87 @@ impl<'d> SpiMasterDriver<'d> {
         let flush_result = bus.flush();
 
         core::mem::drop(lock);
-
         let result = trans_result?;
         finish_result?;
         flush_result?;
 
         Ok(result)
     }
 
-    fn lock_bus(&mut self) -> Result<Lock, EspError> {
-        Lock::new(self.device)
+    pub fn transfer(&self, read: &mut [u8], write: &[u8]) -> Result<(), EspError> {
+        self.transaction(|bus| bus.transfer(read, write))
     }
-}
 
-impl<'d> Drop for SpiMasterDriver<'d> {
-    fn drop(&mut self) {
-        esp!(unsafe { spi_bus_remove_device(self.device) }).unwrap();
-        esp!(unsafe { spi_bus_free(self.host()) }).unwrap();
+    pub fn write(&self, write: &[u8]) -> Result<(), EspError> {
+        self.transaction(|bus| bus.write(write))
     }
-}
 
-unsafe impl<'d> Send for SpiMasterDriver<'d> {}
+    pub fn read(&self, read: &mut [u8]) -> Result<(), EspError> {
+        self.transaction(|bus| bus.read(read))
+    }
+
+    pub fn transfer_in_place(&self, buf: &mut [u8]) -> Result<(), EspError> {
+        self.transaction(|bus| bus.transfer_in_place(buf))
+    }
 
-impl<'d> embedded_hal::spi::ErrorType for SpiMasterDriver<'d> {
+    fn lock_bus(handle: spi_device_handle_t) -> Result<Lock, EspError> {
```

Comment:
Nit: you are always calling `lock_bus` with `self.handle` so a more logical change is to perhaps change the sigunature to `lock_bus(&self)` instead.

Reply:
done